### DISCUSSION TITLE
Syphon / Spout support

### DISCUSSION THREAD

**Discussion by producation** [vcync/modV#927] [Ideas]
Would it be possible to implement Syphon (MacOS) / Spout (Windows) support? It‘s like NDI but shares the video via GPU instead of CPU so it has a few advantages

**Comment by 2xAA** [comment]
No, sorry. There's currently no implementation for either Syphon or Spout for Electron.
I'm keeping an eye out for this but I have been for about 8 years now 😅

**Reply by bennoH2025** [reply]
It's more than a shame that you don't want to take on the work of implementing Spout. NDI is all well and good, but Syphon and especially Spout have become a common standard, particularly in real-time video applications, and so this represents a very large gap in your "modV" program! You should rethink this. Urgently!!

**Reply by 2xAA** [reply]
@bennoH2025 it's not that I don't want to take on the work, it's just that there is no Electron integration for Spout or Syphon currently.

If you know of an implementation I'd be more than happy to look into adding it to modV.

I'm quite aware that it is a limitation of using the web for a program like this.

**Reply by bennoH2025** [reply]
Thank you very much Sam for your quick reply, you really shouldn't have asked me that kind of questions. I'm glad that I'm just about able to build usable ISFs from Shadertoys, but otherwise I have no programming knowledge. Why don't you contact Spout themselves, maybe they have a clever idea. I just think that you'll have to say goodbye to Electron even though you've only just implemented it, but under these circumstances you'll have to find another solution anyway. In my opinion, a real-time video program for VJs and video artists without Spout is worthless, in other words useless. I was able to install your program on Win-10 without any problems, but unfortunately I have to tell you that as far as I can tell with over40 years of experience in the video sector, & overv 25 years as VJ a lot of things don't work there, the structure and workflow take a lot of getting used to, and with a heavy heart you should perhaps consider working with Pixtur from Berlin on Tooll3 for beter.  In my opinion, Tooll3 is really lacking in ISF integration because they opted for the more complex HLSL, which I think is far too complicated for the VJ area, but above all ISF has now become the standard for VJ programs and there are already hundreds & hundrets of great ISF shaders there, and transferring Shadertoys to ISFs often works with not to mutch timeconsuming. 
Your at modV obviously have a lot of experience with ISF and could also develop this area in Tooll3. I also saw that you have already brought some things to Vulkan in modV that are also partly missing in Tooll3 and this program would be worth more clever developers getting involved in in order to make as much progress as possible, which would then bring joy and motivation to everyone involved. I have been observing the dissipation of far too many Linux distributions for years, instead of concentrating on 5 or 6 together and then being able to easily and quickly stand up to Windows and OSX and at least be on par with, or in front of them.  More willingness to compromise and community spirit is what I would like to see in the open source community in general and would be very beneficial to the cause. Unfortunately, I will hardly continue to follow or recommend your program, I'm so sory. But please don't misunderstand me - I see that you were on the right path and invested a lot of patience and time in an extremely self-sacrificing way and shared everything freely and generously with the community. On the other hand, sometimes we all have to face things and take other, new paths. I would see a merger with the Tooll3 community as a great, sensible option for you on modV.
